internal/compile/compile.go

```diff
@@ -124,25 +109,16 @@ func maskRuleFromAnnotationsSet(as *ast.AnnotationSet, comp *ast.Compiler, rule
 
 	for _, ar := range as.Chain(rule) {
 		ann := ar.Annotations
-		if ann == nil {
+		if ann == nil || ann.Compile == nil {
 			continue
 		}
 		// If the mask_rule key is present, validate and parse it.
-		if maskRule, ok := ann.Custom["mask_rule"]; ok {
-			if s, ok := maskRule.(string); ok {
-				maskPath := s
-				if !strings.HasPrefix(s, "data.") {
-					// If the mask_rule is not a data ref try adding package prefix.
-					maskPath = rule.Module.Package.Path.String() + "." + s
-				}
-				maskRuleRef, err := ast.ParseRef(maskPath)
-				if err != nil {
-					hint := FuzzyRuleNameMatchHint(comp, s)
-					return nil, ast.NewError(invalidMaskRuleCode, ann.Loc(), "mask_rule was not a valid ref: %s", hint)
-				}
-				return maskRuleRef, nil
+		if maskRule := ann.Compile.MaskRule; maskRule != nil {
+			if !maskRule.HasPrefix(ast.DefaultRootRef) {
+				// If the mask_rule is not a data ref try adding package prefix.
```

Comment:
"Do or do not, there is no try" 🐸 😄. There is no fail-state here, right?

Reply:
Yeah stale comment. Updated. Thanks!